generators/release/app.js

```diff
@@ -196,7 +199,9 @@ function findOrCreateRelease(args, gen, callback) {
 function createRelease(args, gen, callback) {
    'use strict';
 
-   let releaseDefName = util.isDocker(args.target) ? `${args.teamProject.name}-Docker-CD` : `${args.teamProject.name}-CD`;
+   let kubeDeployment = util.kubeDeployment(args.target);
+   let releaseDefName = util.isDocker(args.target) ? `${args.teamProject.name}-Docker-CD` : kubeDeployment ? `${args.teamProject.name}-kube-${kubeDeployment}`:`${args.teamProject.name}-CD`;
```

Comment:
we should break this up, it's way too complicated for a ternary

Reply:
I agree

Reply:
This needs a larger refactoring to essentially handle the N cases (N==3 right now). Add a helper function if you need and make a simple switch statement. If you find that you have multiple places where you need to create these switches, you can use a dictionary to map <type of deployment, [templates]>. Otherwise, you will be sprinkling in these if/else everywhere.